packages/sycamore-router/src/router.rs

```diff
@@ -98,8 +100,34 @@ impl Integration for HistoryIntegration {
                                 .unwrap_throw();
                             window().scroll_to_with_x_and_y(0.0, 0.0);
```

Comment:
I don't think so because both query and hash is dependent on pathname already.

Reply:
But right now `QUERY` is not updated when `PATHNAME` is updated and the hook only reads `QUERY` so that value would be out of date right? For instance, if navigating from `/abc?q` to `/efg`, `create_query("q")` would not be updated.